### PR TITLE
[PATCH API-NEXT v3] Use chksum API

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

File: helper/include/odp/helper/chksum.h [hunk 1]
```diff
@@ -44,21 +44,7 @@ typedef enum {
...
-	if (len == 1)
-		sum += *(unsigned char *)buf;
-
-	sum = (sum >> 16) + (sum & 0xFFFF);
-	sum += (sum >> 16);
-	result = ~sum;
-
-	return  (__odp_force odp_u16sum_t) result;
+	return ~odp_chksum_ones_comp16(buffer, len);
 }
```
Comment (muvarov) [inline helper/include/odp/helper/chksum.h]: for this patch I think more clean fix is to drop odph_chksum() completely. There is no need to implement helper function if it's simple follows odp function.

**Comment by muvarov** [comment]
patch 2 is ok.

File: helper/include/odp/helper/chksum.h (hunk 1 above)
Comment (lumag) [reply helper/include/odp/helper/chksum.h]: Well... It is a part of ODP helper API, so we can not drop it easily.

Comment (muvarov) [reply helper/include/odp/helper/chksum.h]: so v2 for this?

Comment (lumag) [reply helper/include/odp/helper/chksum.h]: @muvarov this requires consent, as it will be an API change for header library. @psavol any thoughts?

Comment (muvarov) [reply helper/include/odp/helper/chksum.h]: we do not hold api for helpers. They are free to modify.

Comment (muvarov) [reply helper/include/odp/helper/chksum.h]: @lumag  we discussed this on todays meeting. It looks like nobody uses this helper in their app. So dessision is to drop it and we will take a not in changelog.

**Comment by lumag** [comment]
@muvarov updated

**Comment by Bill-Fischofer-Linaro** [comment]
Looks like Travis is failing most of the tests for v2. Needs rework.

**Comment by lumag** [comment]
@muvarov should be fixed now

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**